### PR TITLE
executeCommands: catch and rethrow error with relevant command.

### LINKED ISSUE #606: Mongo command should be included in error message during "executeAll"
I got the below error message and it took me a little bit to figure out which command was throwing the error. I would prefer a message that included the db and command name (although not the full command with arguments, because that would be too much). For example: `Error while running "db.test.insertMany": E11000 duplicate key error collection: test.scores Failed _id or unique key constraint`

Action: cosmosDB.executeAllMongoCommands
Error type: 11000
Error Message: E11000 duplicate key error collection: test.scores Failed _id or unique key constraint

Version: 0.6.1-alpha
OS: darwin

### PR DESCRIPTION
**Pull request by PrashanthCorp**
Fixes #606

### CODE REVIEW

File: src/mongo/MongoScrapbook.ts [hunk 1]
```diff
@@ -74,7 +74,12 @@ export function getAllCommandsFromTextDocument(document: vscode.TextDocument): M
 
 async function executeCommands(activeEditor: vscode.TextEditor, database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext, commands: MongoCommand[]): Promise<void> {
 	for (let command of commands) {
-		await executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
+		try {
+			await executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
+		} catch (e) {
```
Comment (StephenWeatherford) [inline src/mongo/MongoScrapbook.ts]: In JavaScript anything can be thown, so e.message will not work.  use parseError

File: src/mongo/MongoScrapbook.ts [hunk 2]
```diff
@@ -74,7 +74,12 @@ export function getAllCommandsFromTextDocument(document: vscode.TextDocument): M
...
+			e.message = `${command.text.split('(')[0]}, line ${command.range.start.line + 1}, character ${command.range.start.character + 1}: ${e.message}`;
```
Comment (StephenWeatherford) [inline src/mongo/MongoScrapbook.ts]: The reason for using split is unclear (and very suspect).

Comment (StephenWeatherford) [inline src/mongo/MongoScrapbook.ts]: Show an example output please.

Comment (StephenWeatherford) [inline src/mongo/MongoScrapbook.ts]: Don't need to be so specific with "line" and "column" - I'd suggest xx:yy format like compilers do.  But you should review with Matt.

Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: It arose out of `command.collection` not always being defined. I wanted the command prefix that would hold for both of : 
1. `db.collection1.insertMany`
2. `db.getCollectionInfos()`

It seems like printing the substring till the first open bracket is a simple way to extract a well-represented prefix for both cases. 

I myself don't find it cleanest or the most obvious, I'm open to suggestions.

Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: Here: 

db.testcase0.insertMany, line 0, character 0: E11000 duplicate key error collection: trial00.testcase0 Failed _id or unique key constraint

File: src/mongo/MongoScrapbook.ts (hunk 1 above)
Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: Thanks for the catch!

File: src/mongo/MongoScrapbook.ts [hunk 3]
```diff
@@ -77,8 +77,9 @@ async function executeCommands(activeEditor: vscode.TextEditor, database: IAzure
 		try {
 			await executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
 		} catch (e) {
-			e.message = `${command.text.split('(')[0]}, line ${command.range.start.line + 1}, character ${command.range.start.character + 1}: ${e.message}`;
-			throw e;
+			const err = parseError(e);
```
[the remark this answers is not in the document]
Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: This is the output now thrownm with parseError :
db.testcase0.insertMany, 1:1 - E11000 duplicate key error collection: trial00.testcase0 Failed _id or unique key constraint

parseError throws an `IParseError`, guaranteed to have a message field, which is what's displayed when we throw an error. Am I missing something?

[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply src/mongo/MongoScrapbook.ts]: But it's not an error.  new Error(parseError(e).message)

**Comment by StephenWeatherford** [comment]
image

**Pull request merged**